rubix-core/src/main/java/com/qubole/rubix/core/CachingInputStream.java

```diff
@@ -81,7 +82,7 @@
   private byte[] affixBuffer;
   private int diskReadBufferSize;
   private int bufferSize;
-  private BookKeeperFactory bookKeeperFactory;
+  BookKeeperFactory bookKeeperFactory;
```

Comment:
This can be private

Reply:
Required in DummyModeCachingInputStream

Reply:
@garvit-gupta  we are passing the `bookKeeperFactory` object in DummyModeCachingInputStream, so do we still need to be protected ?